### PR TITLE
Add optimizer to truffle-config + update truffle plugins

### PR DESCRIPTION
**Pull request by poiresel**
## Description

Quick description:
- [X] New features

## Changes

Two quick changes
1. Update the truffle-verify + truffle-flatten to the most recent versions
2. Add the optimizer to the truffle-config.js, Test were run using https://github.com/cgewecke/eth-gas-reporter but not significant reductions were seen. Will look to benchmark in test and mainnet deploys though.

## Tests

N/A

## Test Coverage

N/A

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,8 +33,8 @@
     "solhint": "^2.1.0",
     "truffle": "^5.0.32",
     "ganache-time-traveler": "^1.0.5",
-    "truffle-flatten": "^1.0.5",
-    "truffle-verify": "^1.0.3",
+    "truffle-flatten": "^1.0.6",
+    "truffle-verify": "^1.0.4",
```
Comment (alicevhenshaw) [inline package.json]: shall we remove hats? to fix the version and increase it manually?

Comment (poiresel) [reply package.json]: can do

File: source/delegate-factory/truffle-config.js [hunk 2]
```diff
@@ -30,6 +30,12 @@ module.exports = {
   compilers: {
     solc: {
       version: '0.5.12',
+      settings: {
+        optimizer: {
+          enabled: true,
+          runs: 20000,
+        },
+      },
       optimization: false,
```
Comment (alicevhenshaw) [inline source/delegate-factory/truffle-config.js]: Should this read `optimization: false`?

Comment (alicevhenshaw) [reply source/delegate-factory/truffle-config.js]: I see youve now removed this anyway

**Comment by alicevhenshaw** [comment]
Should this be added for `delegate` and `index` too?

File: package.json (hunk 1 above)
Comment (poiresel) [reply package.json]: i don't think so, did the upgrade for visibility but based on tests yarn will upgrade if needed.

**Pull request merged**